Solutions/Menes.Hosting/Menes/Internal/OpenApiOperationInvoker.cs

```diff
@@ -79,6 +79,13 @@ public async Task<TResponse> InvokeAsync(string path, string method, TRequest re
             try
             {
                 IDictionary<string, object> namedParameters = await this.BuildRequestParametersAsync(request, operationPathTemplate).ConfigureAwait(false);
+
+                // Try to find the tenantID in the parameters, but before we check the access policies.
+                if (namedParameters.TryGetValue("tenantId", out object tenantIdObject) && tenantIdObject is string tenantId)
```

Comment:
This has me wondering: are we expecting sometimes to find a named parameter called `tenantId` that isn't a `string`?

My reading of this is that we think it's not problematic if there's no `tenantId` parameter, and also that it's OK if there is one that's something other than a `string`. If that's not the case, then that should be a hard cast so that it a) embodies the expectation that  if `tenantId` is present then it should be a `string`, and b) we will found out if that expectation is misplaced because it will crash visibly, rather than failing silently.

Reply:
We are providing a mechanism that looks for a string called tenantId in the parameters (wherever it came for) and plumbing that in to the context if we find it, and not if we don't. So the criteria are

1. it is called tenantId
2. it is a string

I am not wedded to embedding this convention into the pipeline, and want it removed ASAP. Hence issue #4.